Hazel/Dtls/DtlsUnityConnection.cs

```diff
@@ -89,7 +100,7 @@ struct NextEpoch
         private NextEpoch nextEpoch;
         private TimeSpan handshakeResendTimeout = TimeSpan.FromMilliseconds(200);
 
-        private readonly List<ByteSpan> queuedApplicationData = new List<ByteSpan>();
+        private readonly ConcurrentQueue<QueuedAppData> queuedApplicationData = new ConcurrentQueue<QueuedAppData>();
```

Comment:
Had a question here---is this part of the library run off of the Unity main thread? (or is that a possible use case?). Just checking if the use of Concurrent was intentional or if we can switch to a standard Queue

Reply:
Yeah, this is run off the Unity main thread, but because of the syncLock, technically we could use a normal Queue. Haha, I just like the syntax of the TryDequeue better. Maybe I should just make an extension method for the normal Queue...

Reply:
Haha yeah, I feel you on the TryDequeue---every time I use a normal queue I feel it missing!

And cool, I don't think it'll be an issue either way, but might be good measure to change / add the extension method